README.md

```diff
@@ -2,19 +2,115 @@
 
 > Test client for Xo-Server
 
-## Usage
+## Adding a test
+
+### Organization
+
+```
+src
+├─ user
+|   ├─ __snapshots__
+|   |     └─ index.spec.js.snap 
+|   └─ index.spec.js
+├─ job
+¦   └─ index.spec.js
+¦
+¦
+├─ _xoWithTestHelpers.js
+└─ util.js
+```
 
-xo-server needs to be installed on `localhost:9000`.
-You should also have a `admin@admin.net` account with password `admin`.
+The tests can describe xo methods or scenarios:  
+```javascript
+import { xo } from "../util";
 
-There should be a Xen Server available at `192.168.1.3`with the credentials `root` and password `qwerty`.
+describe("user", () => {
 
-Warning ! This client will create and delete a lot of items (users, groups, servers ...) to test features
+  // testing a method
+  describe(".set()", () => {
+    it("sets an email", async () => {
+      // some tests using xo methods and helpers from _xoWithTestHelpers.js
+      id = await xo.createUser(SIMPLE_USER);
+      expect(await xo.call("user.set", params)).toBe(true);
+      expect(await xo.getUser(id)).toMatchSnapshot({
+        id: expect.any(String),
+      });
+    });
+  });
+      
+  // testing a scenario
+  test("create two users, modify a user email to be the same with the other and fail trying to connect them", () => {
+    /* some tests */
+  });
 
+});
 ```
-> npm ci
-> npm test
-```
+
+### Best practices
+
+- The test environment must remain the same before and after each test:  
+  * each resource created must be deleted  
+  * existing resources should not be altered
+
+- Make a sentence for the title of the test. It must be clear and consistent.
+
+- If the feature you want to test is not implemented : write it and skip it, using `it.skip()`.
+
+- Take values ​​that cover the maximum of testing possibilities.
+
+- If you make tests which keep track of large object, it is better to use snapshots.  
+
+- `_xoWithTestHelpers.js` contains helpers to create temporary resources and to interface with XO.  
```

Comment:
It may be necessary to update to latest changes.

Reply:
`_xoWithTestHelpers.js` -> `_xoConnection`